src/main/kotlin/no/nav/familie/ba/sak/kjerne/steg/JournalførVedtaksbrev.kt

```diff
@@ -43,78 +39,98 @@ class JournalførVedtaksbrev(
         } else {
-            mutableListOf(MottakerInfo(vedtak.behandling.fagsak.aktør.aktivFødselsnummer(), BrukerIdType.FNR, false))
+            val brevMottakere = brevmottakerService.hentBrevmottagere(behandling.id)
+            if (brevMottakere.isNotEmpty()) {
+                mottakere += brevmottakerService.lagMottakereFraBrevMottakere(
+                    brevMottakere,
+                    søkersident,
+                    hentMottakerNavn(søkersident)
+                )
+            } else {
+                mottakere += MottakerInfo(søkersident, BrukerIdType.FNR, false)
+            }
         }
-        if (vedtak.behandling.verge?.ident != null) {
-            mottaker.add(MottakerInfo(vedtak.behandling.verge.ident, BrukerIdType.FNR, true))
+        if (institusjonVergeIdent != null) { // brukes kun i institusjon
+            mottakere += MottakerInfo(vedtak.behandling.verge.ident, BrukerIdType.FNR, true)
         }
 
         val journalposterTilDistribusjon = mutableMapOf<String, MottakerInfo>()
-        mottaker.forEach { mottakerInfo ->
+        mottakere.forEach { mottakerInfo ->
             journalførVedtaksbrev(
                 fnr = fagsak.aktør.aktivFødselsnummer(),
                 fagsakId = fagsakId,
                 vedtak = vedtak,
-                journalførendeEnhet = behanlendeEnhet,
-                mottakerInfo = mottakerInfo
+                journalførendeEnhet = behandlendeEnhet,
+                mottakerInfo = mottakerInfo,
+                tilManuellMottakerEllerVerge = if (institusjonVergeIdent != null) {
+                    mottakerInfo.erInstitusjonVerge
+                } else {
+                    (mottakerInfo.navn != null && mottakerInfo.navn != hentMottakerNavn(søkersident))
+                } // mottakersnavn fyller ut kun når manuell mottaker finnes
             ).also { journalposterTilDistribusjon[it] = mottakerInfo }
         }
 
+        lagTaskForÅDistribuereVedtaksbrev(journalposterTilDistribusjon, data, behandling)
+
+        return hentNesteStegForNormalFlyt(behandling)
+    }
+
+    private fun lagTaskForÅDistribuereVedtaksbrev(
+        journalposterTilDistribusjon: Map<String, MottakerInfo>,
+        data: JournalførVedtaksbrevDTO,
+        behandling: Behandling
+    ) {
         journalposterTilDistribusjon.forEach {
-            if (it.value.erVerge) {
-                val distribuerTilVergeTask = DistribuerVedtaksbrevTilVergeTask.opprettDistribuerVedtaksbrevTilVergeTask(
-                    distribuerVedtaksbrevTilVergeDTO = DistribuerVedtaksbrevTilVergeDTO(
-                        behandlingId = vedtak.behandling.id,
-                        personIdent = it.value.brukerId,
-                        journalpostId = it.key
-                    ),
-                    properties = data.task.metadata
-                )
+            val finnesBrevMottaker = it.value.navn != null &&
+                it.value.navn != hentMottakerNavn(behandling.fagsak.aktør.aktivFødselsnummer())
```

Comment:
```suggestion
            val finnesBrevMottaker =  it.value.navn != hentMottakerNavn(behandling.fagsak.aktør.aktivFødselsnummer())
```
Lurer på om du slipper å sjekke for null her, da det allerede dekkes av andre sjekken som kommer gitt at hentMottakerNavn ikke kan returnere nullable string.

Reply:
fikset